tests/integration_tests/build/test_unittests.py

```diff
@@ -31,6 +31,11 @@ def test_unittests(test_session_root_path, target):
     if "musl" in target:
         extra_env += "TARGET_CC=musl-gcc"
 
+        # On x86_64 with musl target unit tests are already run
+        # as part of testing code-coverage, skipping.
+        if MACHINE == "x86_64":
```

Comment:
nit: we can do this with skipif decorator so we don't have so many if conditions in the test itself; skipif would be something like `target == "musl" and MACHINE = "x86_64"` with `reason = "On x86_64 with musl target unit tests are already run as part of testing code-coverage." `

Reply:
I don't think I can.
We can check for `MACHINE` as that's a global but there is no access to `target` as `target` is a parametrized function param.

Am I missing some python magic I can apply?

Reply:
Yeah, the skipif decorator doesn't take parameters from the parametrize decorator. I didn't realize it.
We could use `pytest.skip()` instead of return here as it is recommended in the [pytest docs](http://doc.pytest.org/en/latest/skipping.html#skipping-test-functions).

Reply:
Done, have a look.